csharp/src/Ice/Connection.cs

```diff
@@ -104,15 +107,7 @@ public TimeSpan IdleTimeout
         /// <summary>The protocol used by the connection.</summary>
         public Protocol Protocol => Endpoint.Protocol;
 
-        // The connector from which the connection was created. This is used by the outgoing connection factory.
-        internal IConnector? Connector => _connector;
-
-        // The endpoints which are associated with this connection. This is populated by the outgoing connection
-        // factory when an endpoint resolves to the same connector as this connection's connector. Two endpoints
-        // can be different but resolve to the same connector (e.g.: endpoints with the IPs "::1", "0:0:0:0:0:0:0:1"
-        // or "localhost" are different endpoints but they all end up resolving to the same connector and can use
-        // the same connection).
-        internal List<Endpoint> Endpoints { get; }
+        internal NonSecure PreferNonSecure { get; }
```

Comment:
Do we need this property? It's unclear what it means for a connection - a connection, once established/created, is either secure or non-secure.

Reply:
We need this to remove the connection from the active connection map.

Reply:
Ok, please add a comment!

Reply:
See other comment - it would be nicer to remove it. It's confusing, and only meaningful for outgoing connections.